datastore/README.md

```diff
@@ -0,0 +1,76 @@
+General-purpose key-value storage.
+The keys are strings, and the values are of any type you want.
+
+> **Note**: This crate is meant to be a utility for the implementation of other crates ; it
+>			does not directly participate in the stack of libp2p.
+
+This crate provides the `Datastore` trait, whose template parameter is the type of the value.
+It is implemented on types that represent a key-value storage.
+The only available implementation for now is `JsonFileDatastore`.
+
+# JSON file datastore
+
+The `JsonFileDatastore` can provide a key-value storage that loads and stores data in a single
+JSON file. It is only available if the value implements the `Serialize`, `DeserializeOwned`
+and `Clone` traits.
+
+The `JsonFileDatastore::new` method will attempt to load existing data from the path you pass
+as parameter. This path is also where the data will be stored. The content of the store is
+flushed on destruction or if you call `flush()`.
+
+```rust
+use datastore::Datastore;
+use datastore::JsonFileDatastore;
```

Comment:
This isn't part of this PR but it'd probably be nice to have this be `datastore::Json`, we don't need to repeat ourselves.

Reply:
I prefer `JsonFileDatastore`, so that when it is used in a third-party crate we know exactly what it is. `Json` alone could make you think that it's just a JSON document in memory.